packages/client/src/transaction/operations.ts

```diff
@@ -40,18 +47,26 @@ export type DeleteTransactionOperation = {
   failIfMissing?: boolean;
 };
 
+export type GetTransactionOperation<O extends XataRecord> = {
+  id: string;
+  columns?: SelectableColumn<O>[];
+};
+
 type TransactionOperationSingleResult<
   Schema extends Record<string, BaseData>,
-  Table extends StringKeys<Schema>,
-  Operation extends TransactionOperation<Schema, Table>
-> = Operation extends { insert: { table: Table; record: { id: infer Id } } }
+  Tables extends StringKeys<Schema>,
+  Operation extends TransactionOperation<Schema, Tables>
+> = Operation extends { insert: { table: Tables; record: { id: infer Id } } }
   ? { operation: 'insert'; id: Id; rows: number }
-  : Operation extends { insert: { table: Table } }
+  : Operation extends { insert: { table: Tables } }
   ? { operation: 'insert'; id: string; rows: number }
-  : Operation extends { update: { table: Table; id: infer Id } }
+  : Operation extends { update: { table: Tables; id: infer Id } }
   ? { operation: 'update'; id: Id; rows: number }
-  : Operation extends { delete: { table: Table } }
+  : Operation extends { delete: { table: Tables } }
   ? { operation: 'delete'; rows: number }
+  : Operation extends { get: { table: infer Table extends Tables } }
+  ? // TODO: Column inference is lost in this case. We should fix this in the future.
```

Comment:
It's not trivial to accomplish this, since we are reworking inference for links, I don't want to jump into this rabbit hole right now.